backend/plugin/db/mysql/mysql.go

```diff
@@ -167,10 +167,59 @@ func registerRDSMysqlCerts(ctx context.Context) error {
 
 	rootCertPool := x509.NewCertPool()
 	if ok := rootCertPool.AppendCertsFromPEM(pem); !ok {
+		return errors.Errorf("failed to parse RDS CA certificates")
+	}
+
+	// Register both secure and insecure configs for backward compatibility
+	// "rds" - maintains backward compatibility (no verification)
+	if err := mysql.RegisterTLSConfig("rds", &tls.Config{
+		RootCAs:            rootCertPool,
+		InsecureSkipVerify: true, // Backward compatible - doesn't verify
+	}); err != nil {
+		return errors.Wrap(err, "failed to register rds TLS config")
+	}
+
+	// "rds-verify" - new secure config that verifies certificates
+	rdVerifyConfig := &tls.Config{
+		RootCAs:            rootCertPool,
+		InsecureSkipVerify: false, // Secure - verifies certificates
+	}
+
+	// Add custom verification function to properly handle intermediate certificates
+	rdVerifyConfig.VerifyPeerCertificate = func(rawCerts [][]byte, _ [][]*x509.Certificate) error {
+		if len(rawCerts) == 0 {
```

Comment:
this code looks complicated and similar to what's inside ssl.go. Can we reuse some code?